src/serac/physics/tests/serac_solid_functional.cpp

```diff
@@ -364,12 +364,120 @@ void functional_parameterized_solid_test(double expected_disp_norm)
+      FiniteElementState::Options{.order = p, .vector_dim = dim, .name = "parameterized_shape"}));
+
+  // Project either a non-affine transformation with an affine transformation on the boundary or
+  // a simple translation on the shape velocity field.
+  if (move_nodes) {
+    mfem::VectorFunctionCoefficient shape_coef(dim, [](const mfem::Vector& x, mfem::Vector& shape_velocity) {
+      shape_velocity[0] = 1.0 + x[0] * (1.0 - x[0]) * x[1] * (1.0 - x[1]) * 2.0;
+      shape_velocity[1] = 1.0 + 0.0;
+    });
+
+    user_defined_shape_velocity.project(shape_coef);
+
+  } else {
+    user_defined_shape_velocity = 1.0;
+  }
+
+  // Save the index of the shape velocity field
+  constexpr int SHAPE_FIELD = 0;
+
+  // Construct a functional-based solid mechanics solver including references to the shape velocity field.
+  SolidFunctional<p, dim, SHAPE_FIELD, H1<p, dim>> solid_solver(default_static, GeometricNonlinearities::On,
+                                                                FinalMeshOption::Reference, "solid_functional",
+                                                                {user_defined_shape_velocity});
+
+  solid_util::NeoHookeanSolid<dim> mat(1.0, 1.0, 1.0);
+  solid_solver.setMaterial(mat);
+
+  // Define the function for the initial displacement and boundary condition
+  auto bc = [](const mfem::Vector&, mfem::Vector& bc_vec) -> void { bc_vec = 0.0; };
+
+  // Set the initial displacement and boundary condition
+  solid_solver.setDisplacementBCs(ess_bdr, bc);
+  solid_solver.setDisplacement(bc);
+
+  // Construct and apply a uniform body load
+  tensor<double, dim> constant_force;
+
+  constant_force[0] = 0.0;
+  constant_force[1] = 1.0e-3;
+
+  if (dim == 3) {
+    constant_force[2] = 0.0;
+  }
+
+  solid_util::ConstantBodyForce<dim> force{constant_force};
+  solid_solver.addBodyForce(force);
+
+  // Finalize the data structures
+  solid_solver.completeSetup();
+
+  // Save the output for inspection
+  solid_solver.initializeOutput(serac::OutputType::VisIt, "shape_optimization");
+
+  // Output the sidre-based plot files
+  solid_solver.outputState();
+
+  // Perform the quasi-static solve
+  double dt = 1.0;
+  solid_solver.advanceTimestep(dt);
+
+  // Output the sidre-based plot files
+  solid_solver.outputState();
+
+  // Check the final displacement norm
+  EXPECT_NEAR(expected_disp_norm, norm(solid_solver.displacement()), 1.0e-5);
 }
 
 TEST(solid_functional, 2D_linear_static) { functional_solid_test_static<1, 2>(1.511052595); }
 TEST(solid_functional, 2D_quad_static) { functional_solid_test_static<2, 2>(2.18604855); }
 TEST(solid_functional, 2D_quad_parameterized_static) { functional_parameterized_solid_test<2, 2>(2.18604855); }
+TEST(solid_functional, 2D_quad_shape_static) { functional_shape_solid_test<2, 2>(0.00011278491, false); }
+TEST(solid_functional, 2D_quad_shape_static_move_nodes) { functional_shape_solid_test<2, 2>(0.000112885, true); }
```

Comment:
Devil's advocate: do we have reason to believe these answers are correct?

Reply:
The moved node calculation is correct when compared to the original one. However, this is not an analytical solution test (which we need).

Reply:
I'm not sure why these answers are only in agreement to a few sig figs, it seems like the shape displacement is a biquadratic function (which should be exactly approximated by the shape displacement function space). I wouldn't expect floating point arithmetic reordering alone to result in such a noticeable difference.

Also: the argument name "move_nodes" for these tests confused me initially, it looks like they both have nonzero shape displacements and yet neither is actually moving the nodes of the mesh (?). 

It might be a good test to compare answers from: 
1. a mesh with a perturbation applied directly to the mesh nodes (evaluated with zero shape displacement)
2. a mesh with no modification to the nodal positions, but evaluated with a shape displacement consistent with the nodal perturbation in 1